[crypo] build fails when `PLATFORM_KEY_REFERENCES_ENABLE` set to `1`

**Describe the bug**

Build fails when `OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE` is set to `1`.

Issue is converting a `const` to a `non-const`:

https://github.com/openthread/openthread/blob/57ef721eeed2550835c528ab7b8e77671f7039d6/src/lib/spinel/radio_spinel.cpp#L893-L898

**To Reproduce**

1. Git commit id 7568e31c0831fcf90fdbe40680f82dc462bde33f
2. IEEE 802.15.4 hardware platform EFR32
3. Build steps https://github.com/openthread/ot-efr32/actions/runs/6478948918/job/17591559259?pr=703
4. Network topology N/A

**Expected behavior**

Build checks should succeed.

**Console/log output**

From:

```
In file included from /__w/ot-efr32/ot-efr32/openthread/src/core/common/data.hpp:43:0,
                 from /__w/ot-efr32/ot-efr32/openthread/src/core/mac/mac_types.hpp:47,
                 from /__w/ot-efr32/ot-efr32/openthread/src/core/mac/mac_frame.hpp:45,
                 from /__w/ot-efr32/ot-efr32/openthread/src/core/radio/radio.hpp:46,
                 from /__w/ot-efr32/ot-efr32/openthread/src/core/radio/max_power_table.hpp:32,
                 from /__w/ot-efr32/ot-efr32/openthread/src/lib/spinel/radio_spinel.hpp:40,
                 from /__w/ot-efr32/ot-efr32/openthread/src/lib/spinel/radio_spinel.cpp:34:
/__w/ot-efr32/ot-efr32/openthread/src/lib/spinel/radio_spinel.cpp: In member function 'otError ot::Spinel::RadioSpinel::SetMacKey(uint8_t, uint8_t, const otMacKeyMaterial*, const otMacKeyMaterial*, const otMacKeyMaterial*)':
/__w/ot-efr32/ot-efr32/openthread/src/lib/spinel/radio_spinel.cpp:893:109: error: invalid conversion from 'const uint8_t* {aka const unsigned char*}' to 'uint8_t* {aka unsigned char*}' [-fpermissive]
     SuccessOrExit(error = otPlatCryptoExportKey(aPrevKey->mKeyMaterial.mKeyRef, aPrevKey->mKeyMaterial.mKey.m8,
                                                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/__w/ot-efr32/ot-efr32/openthread/src/core/common/code_utils.hpp:107:14: note: in definition of macro 'SuccessOrExit'
         if ((aStatus) != 0)    \
              ^~~~~~~
In file included from /__w/ot-efr32/ot-efr32/openthread/include/openthread/platform/radio.h:43:0,
                 from /__w/ot-efr32/ot-efr32/openthread/src/lib/spinel/radio_spinel.hpp:37,
                 from /__w/ot-efr32/ot-efr32/openthread/src/lib/spinel/radio_spinel.cpp:34:
/__w/ot-efr32/ot-efr32/openthread/include/openthread/platform/crypto.h:284:9: note:   initializing argument 2 of 'otError otPlatCryptoExportKey(otCryptoKeyRef, uint8_t*, size_t, size_t*)'
 otError otPlatCryptoExportKey(otCryptoKeyRef aKeyRef, uint8_t *aBuffer, size_t aBufferLen, size_t *aKeyLen);
         ^~~~~~~~~~~~~~~~~~~~~
In file included from /__w/ot-efr32/ot-efr32/openthread/src/core/common/data.hpp:43:0,
                 from /__w/ot-efr32/ot-efr32/openthread/src/core/mac/mac_types.hpp:47,
                 from /__w/ot-efr32/ot-efr32/openthread/src/core/mac/mac_frame.hpp:45,
                 from /__w/ot-efr32/ot-efr32/openthread/src/core/radio/radio.hpp:46,
                 from /__w/ot-efr32/ot-efr32/openthread/src/core/radio/max_power_table.hpp:32,
                 from /__w/ot-efr32/ot-efr32/openthread/src/lib/spinel/radio_spinel.hpp:40,
                 from /__w/ot-efr32/ot-efr32/openthread/src/lib/spinel/radio_spinel.cpp:34:
/__w/ot-efr32/ot-efr32/openthread/src/lib/spinel/radio_spinel.cpp:895:109: error: invalid conversion from 'const uint8_t* {aka const unsigned char*}' to 'uint8_t* {aka unsigned char*}' [-fpermissive]
     SuccessOrExit(error = otPlatCryptoExportKey(aCurrKey->mKeyMaterial.mKeyRef, aCurrKey->mKeyMaterial.mKey.m8,
                                                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/__w/ot-efr32/ot-efr32/openthread/src/core/common/code_utils.hpp:107:14: note: in definition of macro 'SuccessOrExit'
         if ((aStatus) != 0)    \
              ^~~~~~~
In file included from /__w/ot-efr32/ot-efr32/openthread/include/openthread/platform/radio.h:43:0,
                 from /__w/ot-efr32/ot-efr32/openthread/src/lib/spinel/radio_spinel.hpp:37,
                 from /__w/ot-efr32/ot-efr32/openthread/src/lib/spinel/radio_spinel.cpp:34:
/__w/ot-efr32/ot-efr32/openthread/include/openthread/platform/crypto.h:284:9: note:   initializing argument 2 of 'otError otPlatCryptoExportKey(otCryptoKeyRef, uint8_t*, size_t, size_t*)'
 otError otPlatCryptoExportKey(otCryptoKeyRef aKeyRef, uint8_t *aBuffer, size_t aBufferLen, size_t *aKeyLen);
         ^~~~~~~~~~~~~~~~~~~~~
In file included from /__w/ot-efr32/ot-efr32/openthread/src/core/common/data.hpp:43:0,
                 from /__w/ot-efr32/ot-efr32/openthread/src/core/mac/mac_types.hpp:47,
                 from /__w/ot-efr32/ot-efr32/openthread/src/core/mac/mac_frame.hpp:45,
                 from /__w/ot-efr32/ot-efr32/openthread/src/core/radio/radio.hpp:46,
                 from /__w/ot-efr32/ot-efr32/openthread/src/core/radio/max_power_table.hpp:32,
                 from /__w/ot-efr32/ot-efr32/openthread/src/lib/spinel/radio_spinel.hpp:40,
                 from /__w/ot-efr32/ot-efr32/openthread/src/lib/spinel/radio_spinel.cpp:34:
/__w/ot-efr32/ot-efr32/openthread/src/lib/spinel/radio_spinel.cpp:897:109: error: invalid conversion from 'const uint8_t* {aka const unsigned char*}' to 'uint8_t* {aka unsigned char*}' [-fpermissive]
     SuccessOrExit(error = otPlatCryptoExportKey(aNextKey->mKeyMaterial.mKeyRef, aNextKey->mKeyMaterial.mKey.m8,
                                                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/__w/ot-efr32/ot-efr32/openthread/src/core/common/code_utils.hpp:107:14: note: in definition of macro 'SuccessOrExit'
         if ((aStatus) != 0)    \
              ^~~~~~~
In file included from /__w/ot-efr32/ot-efr32/openthread/include/openthread/platform/radio.h:43:0,
                 from /__w/ot-efr32/ot-efr32/openthread/src/lib/spinel/radio_spinel.hpp:37,
                 from /__w/ot-efr32/ot-efr32/openthread/src/lib/spinel/radio_spinel.cpp:34:
/__w/ot-efr32/ot-efr32/openthread/include/openthread/platform/crypto.h:284:9: note:   initializing argument 2 of 'otError otPlatCryptoExportKey(otCryptoKeyRef, uint8_t*, size_t, size_t*)'
 otError otPlatCryptoExportKey(otCryptoKeyRef aKeyRef, uint8_t *aBuffer, size_t aBufferLen, size_t *aKeyLen);
         ^~~~~~~~~~~~~~~~~~~~~
[495/719] Building CXX object openthread/src/core/CMakeFiles/openthread-mtd.dir/utils/slaac_address.cpp.obj
[496/719] Building C object openthread/src/lib/spinel/CMakeFiles/openthread-spinel-ncp.dir/spinel.c.obj
ninja: build stopped: subcommand failed.
```

--- a/src/lib/spinel/radio_spinel.cpp
+++ b/src/lib/spinel/radio_spinel.cpp
@@ -884,35 +884,43 @@ otError RadioSpinel::SetMacKey(uint8_t                 aKeyIdMode,
                                const otMacKeyMaterial *aCurrKey,
                                const otMacKeyMaterial *aNextKey)
 {
-    otError error;
-    size_t  aKeySize;
+    otError  error;
+    size_t   aKeySize;
+    otMacKey prevKey;
+    otMacKey currKey;
+    otMacKey nextKey;
 
     VerifyOrExit((aPrevKey != nullptr) && (aCurrKey != nullptr) && (aNextKey != nullptr), error = kErrorInvalidArgs);
 
 #if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
-    SuccessOrExit(error = otPlatCryptoExportKey(aPrevKey->mKeyMaterial.mKeyRef, aPrevKey->mKeyMaterial.mKey.m8,
-                                                sizeof(aPrevKey->mKeyMaterial.mKey.m8), &aKeySize));
-    SuccessOrExit(error = otPlatCryptoExportKey(aCurrKey->mKeyMaterial.mKeyRef, aCurrKey->mKeyMaterial.mKey.m8,
-                                                sizeof(aCurrKey->mKeyMaterial.mKey.m8), &aKeySize));
-    SuccessOrExit(error = otPlatCryptoExportKey(aNextKey->mKeyMaterial.mKeyRef, aNextKey->mKeyMaterial.mKey.m8,
-                                                sizeof(aNextKey->mKeyMaterial.mKey.m8), &aKeySize));
+    SuccessOrExit(error =
+                      otPlatCryptoExportKey(aPrevKey->mKeyMaterial.mKeyRef, prevKey.m8, OT_MAC_KEY_SIZE, &aKeySize));
+    SuccessOrExit(error =
+                      otPlatCryptoExportKey(aCurrKey->mKeyMaterial.mKeyRef, currKey.m8, OT_MAC_KEY_SIZE, &aKeySize));
+    SuccessOrExit(error =
+                      otPlatCryptoExportKey(aNextKey->mKeyMaterial.mKeyRef, nextKey.m8, OT_MAC_KEY_SIZE, &aKeySize));
 #else
     OT_UNUSED_VARIABLE(aKeySize);
+
+    memcpy(prevKey.m8, aPrevKey->mKeyMaterial.mKey.m8, OT_MAC_KEY_SIZE);
+    memcpy(currKey.m8, aCurrKey->mKeyMaterial.mKey.m8, OT_MAC_KEY_SIZE);
+    memcpy(nextKey.m8, aNextKey->mKeyMaterial.mKey.m8, OT_MAC_KEY_SIZE);
 #endif
 
     SuccessOrExit(error = Set(SPINEL_PROP_RCP_MAC_KEY,
                               SPINEL_DATATYPE_UINT8_S SPINEL_DATATYPE_UINT8_S SPINEL_DATATYPE_DATA_WLEN_S
                                   SPINEL_DATATYPE_DATA_WLEN_S SPINEL_DATATYPE_DATA_WLEN_S,
-                              aKeyIdMode, aKeyId, aPrevKey->mKeyMaterial.mKey.m8, sizeof(otMacKey),
-                              aCurrKey->mKeyMaterial.mKey.m8, sizeof(otMacKey), aNextKey->mKeyMaterial.mKey.m8,
-                              sizeof(otMacKey)));
+                              aKeyIdMode, aKeyId, prevKey.m8, OT_MAC_KEY_SIZE, currKey.m8, OT_MAC_KEY_SIZE, nextKey.m8,
+                              OT_MAC_KEY_SIZE));
 
 #if OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_MAX_COUNT > 0
     mKeyIdMode = aKeyIdMode;
     mKeyId     = aKeyId;
-    memcpy(mPrevKey.m8, aPrevKey->mKeyMaterial.mKey.m8, OT_MAC_KEY_SIZE);
-    memcpy(mCurrKey.m8, aCurrKey->mKeyMaterial.mKey.m8, OT_MAC_KEY_SIZE);
-    memcpy(mNextKey.m8, aNextKey->mKeyMaterial.mKey.m8, OT_MAC_KEY_SIZE);
+
+    memcpy(mPrevKey.m8, prevKey.m8, OT_MAC_KEY_SIZE);
+    memcpy(mCurrKey.m8, currKey.m8, OT_MAC_KEY_SIZE);
+    memcpy(mNextKey.m8, nextKey.m8, OT_MAC_KEY_SIZE);
+
     mMacKeySet = true;
 #endif
 

--- a/src/lib/spinel/radio_spinel.cpp
+++ b/src/lib/spinel/radio_spinel.cpp
@@ -902,9 +902,9 @@ otError RadioSpinel::SetMacKey(uint8_t                 aKeyIdMode,
 #else
     OT_UNUSED_VARIABLE(aKeySize);
 
-    memcpy(prevKey.m8, aPrevKey->mKeyMaterial.mKey.m8, OT_MAC_KEY_SIZE);
-    memcpy(currKey.m8, aCurrKey->mKeyMaterial.mKey.m8, OT_MAC_KEY_SIZE);
-    memcpy(nextKey.m8, aNextKey->mKeyMaterial.mKey.m8, OT_MAC_KEY_SIZE);
+    prevKey = aPrevKey->mKeyMaterial.mKey;
+    currKey = aCurrKey->mKeyMaterial.mKey;
+    nextKey = aNextKey->mKeyMaterial.mKey;
 #endif
 
     SuccessOrExit(error = Set(SPINEL_PROP_RCP_MAC_KEY,
@@ -917,9 +917,9 @@ otError RadioSpinel::SetMacKey(uint8_t                 aKeyIdMode,
     mKeyIdMode = aKeyIdMode;
     mKeyId     = aKeyId;
 
-    memcpy(mPrevKey.m8, prevKey.m8, OT_MAC_KEY_SIZE);
-    memcpy(mCurrKey.m8, currKey.m8, OT_MAC_KEY_SIZE);
-    memcpy(mNextKey.m8, nextKey.m8, OT_MAC_KEY_SIZE);
+    mPrevKey = prevKey;
+    mCurrKey = currKey;
+    mNextKey = nextKey;
 
     mMacKeySet = true;
 #endif

